media-video/ffmpeg/ffmpeg-3.2.6.ebuild

```diff
@@ -300,6 +300,24 @@ src_prepare() {
 	sed -i -e 's/sunos)/sunos) network_extralibs="-lsocket -lnsl"; add_cppflags -D__EXTENSIONS__; enable pic; disable symver ;; no-sunos)/' configure || die
 }
 
+#Check if LDFLAGS contains any LTO flags
+#Must check in reverse!
+#If no LTO flags present, then this is a non-LTO build
+#If the last LTO flag is -fno-lto, that means this is a non-LTO build
+#otherwise, it is an LTO build
+#GCC treats LTO flags in the exact same way
+#Required for bug #631352
+is_lto() {
```

Comment:
I think this function should definitely not be duplicated in every ffmpeg ebuild but rather added to an eclass like flag-o-matic.eclass

Reply:
OK--I'm willing to make a separate pull request for that, but I'm afraid I'm going to need a bit of guidance.  If this is to be extracted into an eclass, like flag-o-matic:

* How general should the function be?  Should it be specific to LTO, specific to LDFLAGS (like `is_ldflagq`), or should there be one variant for LDFLAGS and one for /C/CXX/F/FC flags?

* What name should be given to it (or them, if we have multiple for checking CFLAGS/LDFLAGS/etc)?  We have `is_flagq`--perhaps an `is_flagq_effective` to check if a flag is "effectively" set?  The idea is to view the flags the same way GCC does to toggle that "enable lto" functionality.

* Should `is_flagq` and friends themselves be modified to have this behaviour?  The reason I'm saying this is `is_flagq`totally ignores `-fno-*` style flags.  For example, if you have `CFLAGS=-funroll-loops -fno-unroll-loops`, then `is_flagq -funroll-loops` will happily say `-funroll-loops` is present, even if it is effectively disabled.  However, I could see this change breaking some ebuilds that rely on this...

So yes I agree that this functionality would be useful, I'm just not entirely sure about the right way of going about it.  Could you help me out?

Reply:
You would likely get more useful feedback by posting a patch to gentoo-dev mailing list; this is a general commonly used eclass so that would make sense

What I would do is something like 'is-flag-effective "${FLAGS}" "enabling-flag" "disabling-flag"', so that you can use it like `is-flag-effective "${LDFLAGS}" -flto -fno-lto` for lto or `is-flag-effective "${CFLAGS}" -Werror -Wno-error` & co
Maybe then shortcuts `is-ldflag-effective` based on this.

I would not change the `if-flagq` function as that might have unexpected effects if some ebuild relies on this behavior

Reply:
I had considered that approach, but I realized it's even more troubling than that!  GCC flags can be enabled by other flags, too.  For example, `-O3` enables `-ftree-loop-distribution` on GCC HEAD.  So even having the "enabling flag" and "disabling flag" explicitly listed isn't a catch all either :(

It seems to me that one would actually have to feed GCC your desired CFLAGS and then get an answer from GCC itself about whether a particular flag is enabled.  Wow.  That's quite the stretch from what I originally proposed here!

I've enabled a sort of workaround in my own LTO repo where I strip redundant flags from {C,CXX,LD}FLAGS.  It's not a perfect solution but it does what it needs to do for our use cases there.  I will look at posting to gentoo-dev about this to see what the community thinks about "effective" flags.  Thanks for your help!  I will close this PR and open a new one once a "real" solution comes along.

Reply:
There is also a lazy solution: Add an lto useflag to ffmpeg and let the user figure out if he needs it enabled or not. The problem with the current auto detection is that user does not have a way to say "no, I don't want lto"

Reply:
This actually ended up in flag-o-matic via filter-lto at least, and other modifications landed in is-flag*.